CIScripts/Report/GenerateTestReport.Tests.ps1

```diff
@@ -88,52 +104,57 @@ function Invoke-ReportGenTests {
         }
 
         It "creates appropriate files" {
-            Join-Path $OutputDir "raw_NUnit/foo.xml" | Should Exist
+            Join-Path $OutputDir "raw_NUnit/SomeSuite.xml" | Should Exist
+            Join-Path $OutputDir "pretty_test_report/SomeSuite.html" | Should Exist
+            Join-Path $OutputDir "pretty_test_report/OtherSuite.html" | Should Exist
             Join-Path $OutputDir "pretty_test_report/Index.html" | Should Exist
             Join-Path $OutputDir "reports-locations.json" | Should Exist
         }
 
-        It "flattens the xml files" {
-            $ExpectedXml = NormalizeXmlString '
-            <test-results failures="0" inconclusive="0" skipped="0" date="2018-01-01" time="15:00:00">
-                <test-suite name="inner_suite" type="TestFixture" result="Success">
-                    <results>
-                        <test-case name="test" result="Success" />
-                    </results>
-                </test-suite>
-            </test-results>'
-            $FileContents = Get-Content -Raw (Join-Path $OutputDir "raw_NUnit/foo.xml")
-            NormalizeXmlString $FileContents | Should BeExactly $ExpectedXml
-        }
-
-        Test-JsonFileForMonitoring "foo.xml"
-    }
-
-    Context "multiple xml files" {
-        BeforeAll {
-            $InputDir, $OutputDir = New-TemporaryDirs
-            New-DummyFile -Path (Join-Path $InputDir "foo.xml")
-            New-DummyFile -Path (Join-Path $InputDir "bar.xml")
-            New-DummyFile -Path (Join-Path $InputDir "baz.xml")
-            Convert-TestReportsToHtml -XmlReportsDir $InputDir -OutputDir $OutputDir -GeneratorFunc $ReportunitWrapperFunc
+        It "splits combined xml files and flattens the contents to max 1 level of test suites" {
+            $ExpectedSomeSuiteXml = NormalizeXmlString @"
+<?xml version="1.0"?>
+<test-results>
+<environment />
+<culture-info current-culture="en-US" current-uiculture="en-US" />
+    <test-suite name="1">
+    <results>
+        <test-case description="MyTestCase" name="MyTestCase" />
+    </results>
+    </test-suite>
+</test-results>
+"@
+            $ExpectedOtherSuiteXml = NormalizeXmlString @"
+<?xml version="1.0"?>
+<test-results>
+<environment />
+<culture-info current-culture="en-US" current-uiculture="en-US" />
+    <test-suite name="2">
+    <results>
+        <test-case description="YourTestCase" name="YourTestCase" />
+    </results>
+    </test-suite>
+</test-results>
+"@
+            NormalizeXmlString (Get-Content -Raw (Join-Path $OutputDir "raw_NUnit/SomeSuite.xml")) `
+                | Should BeExactly $ExpectedSomeSuiteXml
+            NormalizeXmlString (Get-Content -Raw (Join-Path $OutputDir "raw_NUnit/OtherSuite.xml")) `
+                | Should BeExactly $ExpectedOtherSuiteXml
         }
 
-        AfterAll {
-            Clear-TemporaryDirs -Dirs @($InputDir, $OutputDir)
+        It "json file for monitoring contains valid path to raw nunit reports" {
+            Param($Filename)
```

Comment:
Is `$Filename` parameter necessary? We can't see any usage of it here.

Question also applies to the next test case.